wrangler-core/src/main/java/io/cdap/directives/column/SetType.java

```diff
@@ -23,43 +23,62 @@
 import io.cdap.wrangler.api.Directive;
 import io.cdap.wrangler.api.DirectiveExecutionException;
 import io.cdap.wrangler.api.ExecutorContext;
+import io.cdap.wrangler.api.Optional;
 import io.cdap.wrangler.api.Row;
 import io.cdap.wrangler.api.annotations.Categories;
 import io.cdap.wrangler.api.lineage.Lineage;
 import io.cdap.wrangler.api.lineage.Mutation;
+import io.cdap.wrangler.api.parser.Bool;
 import io.cdap.wrangler.api.parser.ColumnName;
 import io.cdap.wrangler.api.parser.Identifier;
+import io.cdap.wrangler.api.parser.Numeric;
 import io.cdap.wrangler.api.parser.TokenType;
 import io.cdap.wrangler.api.parser.UsageDefinition;
 import io.cdap.wrangler.utils.ColumnConverter;
 
+import java.math.RoundingMode;
 import java.util.List;
 
 /**
- * A Wrangler step for converting data type of a column
+ * A Wrangler step for converting data type of column
  * Accepted types are: int, short, long, double, float, string, boolean and bytes
+ * When decimal type is selected, can also specify the scale and whether to round or not
  */
 @Plugin(type = "directives")
 @Name(SetType.NAME)
 @Categories(categories = {"column"})
 @Description("Converting data type of a column.")
 public final class SetType implements Directive, Lineage {
   public static final String NAME = "set-type";
+
+  // Default scale that is assumed in CDAP output schema
+  public static final int DEFAULT_SCALE = 3;
+
   private String col;
   private String type;
+  private int scale;
+  private RoundingMode roundingMode;
+
 
   @Override
   public UsageDefinition define() {
     UsageDefinition.Builder builder = UsageDefinition.builder(NAME);
     builder.define("column", TokenType.COLUMN_NAME);
     builder.define("type", TokenType.IDENTIFIER);
+    builder.define("scale", TokenType.NUMERIC, Optional.TRUE);
+    builder.define("round", TokenType.BOOLEAN, Optional.TRUE);
     return builder.build();
   }
 
   @Override
   public void initialize(Arguments args) {
     col = ((ColumnName) args.value("column")).value();
     type = ((Identifier) args.value("type")).value();
+    if (type.equalsIgnoreCase("decimal")) {
+      scale = args.contains("scale") ? ((Numeric) args.value("scale")).value().intValue() : DEFAULT_SCALE;
```

Comment:
Although this is ok for smaller set of arguments. In this case, you are searching the arguments twice for "scale", since both contains and value will search for the input. Instead, you can simply take the value in a param and use that for null (similar to TryGetValue you have in dictionaries). You can choose to ignore this comment, but would be good to put this into practice for any such search pattern.

Reply:
Arguments is implemented as a map here. 'Contains' map to Map's containsKey(https://github.com/data-integrations/wrangler/blob/8be2f88adc4fb69fb8c103301228d32f35bdea96/wrangler-core/src/main/java/io/cdap/wrangler/parser/MapArguments.java#L153) and value maps to map's get function which does not invoke containsKey again.